### PR TITLE
more docs

### LINKED ISSUE #31: Further documentation improvements
The documentation can be improved by:
* Changing the rest of the text from upper-case, so that the docs are not SHOUTING so much.
* I couldn't get automatic links to subroutines working using their upper-case names. So I can use sptrun() and I get a link, but when I change it to SPTRUN() the link does not happen. It would be better to make the upper-case links work, since this is all F77 code and people are used to it being in upper-case.
* Examples can be much better handled.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #31 

I think this is the last round of documentation edits for this release...

### CODE REVIEW

File: src/sppad.f [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-C>                (0 FOR TRIANGULAR, 1 FOR RHOMBOIDAL)
-C> @param M2       - INTEGER OUTPUT SPECTRAL TRUNCATION
+C> @param I1 input spectral domain shape
+C> (0 for triangular, 1 for rhomboidal)
+C> @param M1 input spectral truncation
+C> @param Q1 ((m+1)*((i+1)*m+2)) input field
+C> @param I2 output spectral domain shape
+C> (0 for triangular, 1 for rhomboidal)
+C> @param M2 output spectral truncation
+C> @param Q2 ((M+1)*((I+1)*M+2)) output field
```
Comment (AlexanderRichert-NOAA) [inline src/sppad.f]: Inconsistent capitalization of variables (presumably "m" and "i" should be re-capitalized)

File: src/spsynth.f [hunk 2]
```diff
@@ -1,43 +1,44 @@
 C> @file
+C> @brief Synthesize Fourier coefficients from spectral coefficients.
 C>
-C> Synthesize fourier from spectral
-C> @author IREDELL @date 92-10-31
+C> 91-10-31 | Mark Iredell | Initial.
```
Comment (AlexanderRichert-NOAA) [inline src/spsynth.f:7]: No table header

File: src/spsynth.f [hunk 3]
```diff
@@ -1,43 +1,44 @@
...
+C> (NCTOP>=2*(M+1))
+C> @param KM number of fields
+C> @param CLAT cosine of latitude
+C> @param PLN ((M+1)*((I+1)*M+2)/2) Legendre polynomial
+C> @param PLNTOP Legendre polynomial over top
+C> @param SPC spectral coefficients
+C> @param SPCTOP spectral coefficients over top
+C> @param MP identifiers (0 for scalar, 1 for vector,
+C> or 10 for scalar and gradient)
+C> @param F Fourier coefficients for latitude pair.
```
Comment (AlexanderRichert-NOAA) [inline src/spsynth.f]: Doesn't need period at end

File: src/sptez.f [hunk 4]
```diff
@@ -1,53 +1,56 @@
...
-C>   -----------------------      |---------           |-------------
-C>   IMAX                         |2*MAXWV+2           |3*MAXWV/2*2+2
-C>   JMAX (IDRT=4,IROMB=0)        |1*MAXWV+1           |3*MAXWV/2+1
-C>   JMAX (IDRT=4,IROMB=1)        |2*MAXWV+1           |5*MAXWV/2+1
-C>   JMAX (IDRT=0,IROMB=0)        |2*MAXWV+3           |3*MAXWV/2*2+3
-C>   JMAX (IDRT=0,IROMB=1)        |4*MAXWV+3           |5*MAXWV/2*2+3
-C>   JMAX (IDRT=256,IROMB=0)      |2*MAXWV+1           |3*MAXWV/2*2+1
-C>   JMAX (IDRT=256,IROMB=1)      |4*MAXWV+1           |5*MAXWV/2*2+1
+C> Minimum grid dimensions for unaliased transforms to spectral:
+C> DIMENSION                    |LINEAR              |QUADRATIC
```
Comment (AlexanderRichert-NOAA) [inline src/sptez.f:23]: If it were me, I would remove the ":" from the table title, but in any case we've been de-upper casing the table column names.

File: src/sptez.f [hunk 5]
```diff
@@ -1,53 +1,56 @@
...
+C>
+C> @param IROMB spectral domain shape
+C> (0 for triangular, 1 for rhomboidal)
+C> @param MAXWV spectral truncation
+C> @param IDRT grid identifier
+C> - IDRT=4 for Gaussian grid,
+C> - IDRT=0 for equally-spaced grid including poles,
+C> - IDRT=256 for equally-spaced grid excluding poles
+C> @param IMAX even number of longitudes.
+C> @param JMAX number of latitudes.
```
Comment (AlexanderRichert-NOAA) [inline src/sptez.f:41]: extraneous periods

File: src/sptez.f [hunk 6]
```diff
@@ -1,53 +1,56 @@
...
+C> @param[out] WAVE wave field if IDIR>0
+C> where MX=(MAXWV+1)*((IROMB+1)*MAXWV+2)/2
+C> @param[out] GRID grid field (E->W,N->S) if IDIR<0
+C> @param IDIR transform flag
+C> (IDIR>0 for wave to grid, IDIR<0 for grid to wave).
```
Comment (AlexanderRichert-NOAA) [inline src/sptez.f:46]: " "

File: src/sptezd.f [hunk 7]
```diff
@@ -1,58 +1,60 @@
...
+C> The wave field is in sequential 'IBM ORDER'.
+C>
+C> The grid fiels is indexed East to West, then North to South.
+C>
+C> For more flexibility and efficiency, call sptran().
+C>
+C> Subprogram can be called from a multiprocessing environment.
+C>
+C> Minimum grid dimensions for unaliased transforms to spectral:
+C> DIMENSION                    |LINEAR              |QUADRATIC
```
Comment (AlexanderRichert-NOAA) [inline src/sptezd.f:23]: Same comment as above for sptez.f

File: src/sptezd.f [hunk 8]
```diff
@@ -1,58 +1,60 @@
...
+C> - IDRT=0 for equally-spaced grid including poles
+C> - IDRT=256 for equally-spaced grid excluding poles
+C> @param IMAX even number of longitudes.
+C> @param JMAX number of latitudes.
+C> @param[out] WAVE wave field if IDIR>0
+C> @param[out] GRIDMN global mean if IDIR<0
+C> @param[out] GRIDX grid x-gradients (E->W,N->S) if IDIR<0
+C> @param[out] GRIDY grid y-gradients (E->W,N->S) if IDIR<0
+C> @param IDIR transform flag
+C> (IDIR>0 for wave to grid, IDIR<0 for grid to wave).
```
Comment (AlexanderRichert-NOAA) [inline src/sptezd.f:47]: extra periods (apparently I'm in a nitpicky mood today)

File: src/sptezm.f [hunk 9]
```diff
@@ -1,54 +1,57 @@
...
-C>   -----------------------      |---------           |-------------
-C>   IMAX                         |2*MAXWV+2           |3*MAXWV/2*2+2
-C>   JMAX (IDRT=4,IROMB=0)        |1*MAXWV+1           |3*MAXWV/2+1
-C>   JMAX (IDRT=4,IROMB=1)        |2*MAXWV+1           |5*MAXWV/2+1
-C>   JMAX (IDRT=0,IROMB=0)        |2*MAXWV+3           |3*MAXWV/2*2+3
-C>   JMAX (IDRT=0,IROMB=1)        |4*MAXWV+3           |5*MAXWV/2*2+3
-C>   JMAX (IDRT=256,IROMB=0)      |2*MAXWV+1           |3*MAXWV/2*2+1
-C>   JMAX (IDRT=256,IROMB=1)      |4*MAXWV+1           |5*MAXWV/2*2+1
+C> Minimum grid dimensions for unaliased transforms to spectral:
+C> DIMENSION                    |LINEAR              |QUADRATIC
```
Comment (AlexanderRichert-NOAA) [inline src/sptezm.f:23]: Same comment as above

File: src/sptezm.f [hunk 10]
```diff
@@ -1,54 +1,57 @@
...
+C> - IDRT=0 for equally-spaced grid including poles
+C> - IDRT=256 for equally-spaced grid excluding poles
+C> @param IMAX even number of longitudes
+C> @param JMAX number of latitudes
+C> @param KMAX number of fields to transform
+C> @param[out] WAVE wave field if IDIR>0
+C> where MX=(MAXWV+1)*((IROMB+1)*MAXWV+2)/2
+C> @param[out] GRID grid field (E->W,N->S) if IDIR<0
+C> @param IDIR transform flag
+C> (IDIR>0 for wave to grid, IDIR<0 for grid to wave).
```
Comment (AlexanderRichert-NOAA) [inline src/sptezm.f:47]: period

File: src/sptezmd.f [hunk 11]
```diff
@@ -1,61 +1,63 @@
...
+C> The wave fields are in sequential 'IBM ORDER'.
+C>
+C> The grid fields are indexed East to West, then North to South.
+C>
+C> For more flexibility and efficiency, call sptran().
+C>
+C> Subprogram can be called from a multiprocessing environment.
+C>
+C> Minimum grid dimensions for unaliased transforms to spectral:
+C> DIMENSION                    |LINEAR              |QUADRATIC
```
Comment (AlexanderRichert-NOAA) [inline src/sptezmd.f:23]: Same comment as above

File: src/sptezmd.f [hunk 12]
```diff
@@ -1,61 +1,63 @@
...
+C> @param JMAX number of latitudes.
+C> @param KMAX number
+C> @param[out] WAVE wave field if IDIR>0
+C> where MX=(MAXWV+1)*((IROMB+1)*MAXWV+2)
+C> @param[out] GRIDMN global mean if IDIR<0
+C> @param[out] GRIDX grid x-gradients (E->W,N->S) if IDIR<0
+C> @param[out] GRIDY grid y-gradients (E->W,N->S) if IDIR<0
+C> @param IDIR transform flag
+C> (IDIR>0 for wave to grid, IDIR<0 for grid to wave).
+C>
```
Comment (AlexanderRichert-NOAA) [inline src/sptezmd.f:50]: periods

File: src/sptezmv.f [hunk 13]
```diff
@@ -1,60 +1,63 @@
...
-C>   -----------------------      |---------           |-------------
-C>   IMAX                         |2*MAXWV+2           |3*MAXWV/2*2+2
-C>   JMAX (IDRT=4,IROMB=0)        |1*MAXWV+1           |3*MAXWV/2+1
-C>   JMAX (IDRT=4,IROMB=1)        |2*MAXWV+1           |5*MAXWV/2+1
-C>   JMAX (IDRT=0,IROMB=0)        |2*MAXWV+3           |3*MAXWV/2*2+3
-C>   JMAX (IDRT=0,IROMB=1)        |4*MAXWV+3           |5*MAXWV/2*2+3
-C>   JMAX (IDRT=256,IROMB=0)      |2*MAXWV+1           |3*MAXWV/2*2+1
-C>   JMAX (IDRT=256,IROMB=1)      |4*MAXWV+1           |5*MAXWV/2*2+1
+C> Minimum grid dimensions for unaliased transforms to spectral:
+C> DIMENSION                    |LINEAR              |QUADRATIC
```
Comment (AlexanderRichert-NOAA) [inline src/sptezmv.f:23]: Same comment as above

File: src/sptezmv.f [hunk 14]
```diff
@@ -1,60 +1,63 @@
...
+C> @param JMAX number of latitudes
+C> @param KMAX number of fields to transform
+C> @param[out] WAVED wave divergence field if IDIR<0
+C> where MX=(MAXWV+1)*((IROMB+1)*MAXWV+2)/2
+C> @param[out] WAVEZ wave vorticity field if IDIR>0
+C> where MX=(MAXWV+1)*((IROMB+1)*MAXWV+2)/2
+C> @param[out] GRIDU grid u-wind (E->W,N->S) if IDIR>0
+C> @param[out] GRIDV grid v-wind (E->W,N->S) if IDIR>0
+C> @param IDIR transform flag
+C> (IDIR>0 for wave to grid, IDIR<0 for grid to wave).
```
Comment (AlexanderRichert-NOAA) [inline src/sptezmv.f:50]: period

File: src/sptgpmd.f [hunk 15]
```diff
@@ -1,6 +1,6 @@
 C> @file
 C> @brief Transform spectral to Mercator gradients
-C> @author IREDELL @date 96-02-29
+C> @author Iredell @date 96-02-29
 
 C> This subprogram performs a spherical transform
 C> from spectral coefficients of scalar fields
```
Comment (AlexanderRichert-NOAA) [inline src/sptgpmd.f:6]: punctuation

**Comment by edwardhartnett** [comment]
@AlexanderRichert-NOAA please review and we'll get this merged.

**Comment by AlexanderRichert-NOAA** [comment]
@edwardhartnett Comments are there, just need to be resolved (mainly minor punctuation/capitalization issues)

**Comment by edwardhartnett** [comment]
@AlexanderRichert-NOAA I don't see any review comments.

I seem to remember that there is a button at the bottom of the screen where you have to submit your review. If you don't all comments are lost. 

I usually select "post single comment" instead of review for this reason. ;-)

**Review by AlexanderRichert-NOAA — CHANGES_REQUESTED** [review]

**Comment by AlexanderRichert-NOAA** [comment]
@edwardhartnett That sound in the distance was me slapping my forehead...

File: src/sppad.f (hunk 1 above)
Comment (edwardhartnett) [reply src/sppad.f]: Fixed.

File: src/spsynth.f (hunk 2 above)
Comment (edwardhartnett) [reply src/spsynth.f:7]: Fixed.

File: src/spsynth.f (hunk 3 above)
Comment (edwardhartnett) [reply src/spsynth.f]: OK, but I also frequenly add a period on the end of each param description. THis punctuation-less world of upper-case fortran commenting drives me mad...

File: src/sptez.f (hunk 5 above)
Comment (edwardhartnett) [reply src/sptez.f:41]: Let's not worry about them.

File: src/sptgpmd.f (hunk 15 above)
Comment (edwardhartnett) [reply src/sptgpmd.f:6]: Fixed.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Pull request merged**